app/routes.py

```diff
@@ -1 +1,271 @@
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return make_response(jsonify({
+        "details" : f"Goal {goal_id} \"{goal.title}\" successfully deleted"}))
+
+
+def use_slack_bot(task):
+    PATH = "https://slack.com/api/chat.postMessage"
+    SLACK_API_KEY = os.environ.get("API_KEY")
+    
+    query_params = {
+        "channel": "#task-notifications",
+        "text" : f"Someone just completed the task {task.title}"
+        }
+
+    requests.post(PATH, params=query_params, headers={
+        "Authorization": SLACK_API_KEY
+    })
+
+@bp.route("<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_on_incompleted_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+    use_slack_bot(task)
+    return {"task":task.to_dict()}, 200
+
+    
+
+
+
+
+@bp.route("<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_incomplete_on_completed_task(task_id):
+    try:
+        task = validate_model(Task, task_id)
+        task.completed_at = None
+
+        db.session.commit()
+
+        return make_response(jsonify({
+            "task": Task.to_dict(task)}))
+    except:
+        abort(make_response({"message" : f"task {task_id} not found"}, 404))
+
+@bp.route("<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete_on_completed_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.utcnow()
+    task.is_complete = True
+    task_dict = task.to_dict()
+
+    db.session.commit()
+
+    return make_response(jsonify({
+        "task": task_dict}))
+
+@bp.route("<task_id>/mark_incomplete", methods=["PATCH"])
+
+def mark_incomplete_on_incompleted_task(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = None
+
+    db.session.commit()
+
+    return make_response(jsonify({
+        "task": Task.to_dict(task)}))
+
+
+@goal_bp.route("/<goal_id>/tasks", methods=["POST"])
+def send_list_of_tasks_to_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+    request_body = request.get_json()
+
+    goal.tasks = []
+
+    for task_id in request_body["task_ids"]:
+        task = validate_model(Task, task_id)
+        goal.tasks.append(task)
+
+
+    # db.session.add()
+    db.session.commit()
+
+    return make_response(jsonify({
+        "id" : goal.goal_id,
+        "task_ids" : request_body["task_ids"]
+    })), 200
+
+@goal_bp.route("/<goal_id>/tasks", methods=["GET"])
+
+def get_tasks_of_goal(goal_id):
+    goal = validate_model(Goal, goal_id)
+
+    all_tasks = []
+
+    tasks = goal.tasks
+
+    for task in tasks:
+        all_tasks.append({
+            "id" : task.task_id,
+            "goal_id" : task.goal_id,
+            "title" : task.title,
+            "description" : task.description,
+            "is_complete" : task.is_complete
+        })
+
+    return jsonify({
+        "id" : goal.goal_id,
+        "title" : goal.title,
+        "tasks" : all_tasks
```

Comment:
Could we use your to_dict() method in the Goal class instead of re-writing a dictionary literal here?

Your current implementation doesn't set "tasks" but you could add some conditional logic to to_dict to set "tasks" if needed